### ISSUE TITLE
[nvmf_filesystem] Cannot remove namespace file

### ISSUE THREAD

**Issue by tomzawadzki** [spdk/spdk#1906]
Execution fails in nvmf_tcp_phy autotest, which suggest that changes in patch below might affect it:
https://review.spdk.io/gerrit/c/spdk/spdk/+/6932
@wanghailiangx Do would you mind taking a look ?

## Link to the failed CI build

https://review.spdk.io/gerrit/c/spdk/spdk/+/7561/
https://ci.spdk.io/results/autotest-per-patch/builds/43146/archive/nvmf-tcp-phy-autotest/build.log

## Execution failed at
```
00:02:50.323   17:49:56	# NVMF_TARGET_INTERFACE=enp10s0f0
00:02:50.323   17:49:56	# NVMF_INITIATOR_INTERFACE=enp10s0f1
00:02:50.323   17:49:56	# NVMF_SECOND_TARGET_IP=
00:02:50.323   17:49:56	# NVMF_TARGET_NAMESPACE=enp10s0f0_ns
00:02:50.323   17:49:56	# NVMF_TARGET_NS_CMD=(ip netns exec "$NVMF_TARGET_NAMESPACE")
00:02:50.323   17:49:56	# ip netns del enp10s0f0_ns
**00:02:50.324  Cannot remove namespace file "/var/run/netns/enp10s0f0_ns": No such file or directory**
00:02:50.324   17:49:56	# true
00:02:50.324   17:49:56	# ip -4 addr flush enp10s0f0
00:02:50.324   17:49:56	# ip -4 addr flush enp10s0f1
00:02:50.326   17:49:56	# trap 'nvmf_tcp_fini; exit 1' SIGINT SIGTERM
00:02:50.326   17:49:56	# ip netns add enp10s0f0_ns
00:02:50.330   17:49:56	# ip link set enp10s0f0 netns enp10s0f0_ns
00:02:50.416   17:49:56	# ip addr add 10.0.0.1/24 dev enp10s0f1
00:02:50.424   17:49:56	# ip netns exec enp10s0f0_ns ip addr add 10.0.0.2/24 dev enp10s0f0
00:02:50.428   17:49:56	# ip link set enp10s0f1 up
00:02:50.430   17:49:56	# ip netns exec enp10s0f0_ns ip link set enp10s0f0 up
00:02:50.500   17:49:56	# ip netns exec enp10s0f0_ns ip link set lo up
00:02:50.507   17:49:56	# iptables -I INPUT 1 -i enp10s0f1 -p tcp --dport 4420 -j ACCEPT
00:02:50.517   17:49:56	# ping -c 1 10.0.0.2
00:03:00.520  PING 10.0.0.2 (10.0.0.2) 56(84) bytes of data.
00:03:00.520  
00:03:00.520  --- 10.0.0.2 ping statistics ---
00:03:00.520  1 packets transmitted, 0 received, **100% packet loss**, time 0ms
```

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @tomzawadzki. log: https://ci.spdk.io/public_build/autotest-per-patch_43146.html

**Comment by wanghailiangx** [comment]
From dmesg, we can see the insterface enp10s0f0 go through a restart process.
```
[  516.931866] block nbd1: NBD_DISCONNECT
[  516.931947] block nbd1: Unexpected reply (0) 0000000088dc9e77
[  516.931953] block nbd1: Disconnected due to user request.
[  527.576425] IPv6: ADDRCONF(NETDEV_CHANGE): enp10s0f0: link becomes ready
```

This issue can be reproduced by chance. In this circumstances, the local target inferface ens786f0 is not displayed. 
"ifconfig ens786f0 up" can't work, show:
`ens786f0: ERROR while getting interface flags: No such device`
After half a minute, and there was no operation, ths ens786f0 was up again.

To solve this problem, it didn't work to re install the nvme-tcp driver,
Only when the ice driver is re installed, the network interface can be up immediately.
modprobe -v -r ice
modprobe ice
I put these after nvmf_tcp_fini as a workaround, please see https://review.spdk.io/gerrit/c/spdk/spdk/+/7573
but I'm not sure if this is a driver related issue.

There is, too, in the process of running nvme tcp test such as fio, the IP of a certain interface will be lost. This causes 
```
ctrlr.c: 185:nvmf_ctrlr_keep_alive_poll: NOTICE: Disconnecting host nqn.2014-08.org.nvmexpress:
....... keep alive timeout.
Build timed out (after 15 minutes). Marking the build as failed.
```
This problem can appear every time under these conditions:
**set runtime=60 in examples/nvme/fio_plugin/example_config.fio**
./test/nvmf/host/fio.sh --transport=tcp --iso
```
************************************
START TEST nvmf_fio
************************************
........
 14:58:40       # '[' tcp == tcp ']'
 14:58:40       # modprobe nvme-tcp
 14:58:40       # '[' '' == iso ']'
 14:58:40       # NVMF_TRANSPORT_OPTS='-t tcp'
 14:58:40       # [[ tcp == \r\d\m\a ]]
 14:58:40       # [[ tcp == \t\c\p ]]
 14:58:40       # nvmf_tcp_init
 14:58:40       # NVMF_INITIATOR_IP=10.0.0.1
 14:58:40       # NVMF_FIRST_TARGET_IP=10.0.0.2
 14:58:40       # TCP_INTERFACE_LIST=($(get_tcp_if_list_by_driver))
  14:58:40      # get_tcp_if_list_by_driver
  14:58:40      # local driver
  14:58:40      # driver=ice
  14:58:40      # shopt -s nullglob
  14:58:40      # tcp_if_list=(/sys/bus/pci/drivers/$driver/0000*/net/*)
  14:58:40      # shopt -u nullglob
  14:58:40      # printf '%s\n' ens786f0 ens786f1
 14:58:40       # (( 2 == 0 ))
 14:58:40       # (( 2 > 1 ))
 14:58:40       # NVMF_TARGET_INTERFACE=ens786f0
 14:58:40       # NVMF_INITIATOR_INTERFACE=ens786f1
 14:58:40       # NVMF_SECOND_TARGET_IP=
 14:58:40       # NVMF_TARGET_NAMESPACE=ens786f0_ns
 14:58:40       # NVMF_TARGET_NS_CMD=(ip netns exec "$NVMF_TARGET_NAMESPACE")
 14:58:40       # trap 'nvmf_tcp_fini; exit 1' SIGINT SIGTERM
 14:58:40       # ip netns add ens786f0_ns
 14:58:40       # ip link set ens786f0 netns ens786f0_ns
 14:58:40       # ip addr add 10.0.0.1/24 dev ens786f1
 14:58:40       # ip netns exec ens786f0_ns ip addr add 10.0.0.2/24 dev ens786f0
 14:58:40       # ip link set ens786f1 up
 14:58:40       # ip netns exec ens786f0_ns ip link set ens786f0 up
 14:58:40       # ip netns exec ens786f0_ns ip link set lo up
 14:58:40       # iptables -I INPUT 1 -i ens786f1 -p tcp --dport 4420 -j ACCEPT
 14:58:40       # ping -c 1 10.0.0.2
PING 10.0.0.2 (10.0.0.2) 56(84) bytes of data.
64 bytes from 10.0.0.2: icmp_seq=1 ttl=64 time=0.202 ms

--- 10.0.0.2 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.202/0.202/0.202/0.000 ms
 14:58:40       # ip netns exec ens786f0_ns ping -c 1 10.0.0.1
PING 10.0.0.1 (10.0.0.1) 56(84) bytes of data.
64 bytes from 10.0.0.1: icmp_seq=1 ttl=64 time=0.057 ms

--- 10.0.0.1 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.057/0.057/0.057/0.000 ms
 14:58:40       # NVMF_APP=("${NVMF_TARGET_NS_CMD[@]}" "${NVMF_APP[@]}")
 14:58:40       # NVMF_TRANSPORT_OPTS='-t tcp -o'
 14:58:40       # [[ y != y ]]
 14:58:40       # timing_enter start_nvmf_tgt
 14:58:40       # xtrace_disable
 14:58:40       # '[' '' '!=' yes ']'
 14:58:40       # PREV_BASH_OPTS=ehxBET
 14:58:40       # [[ ehxBET == *\x* ]]
 14:58:40       # XTRACE_DISABLED=yes
 14:58:40       # set +x
 14:58:40       # xtrace_enable
 14:58:40       # nvmfpid=59291
 14:58:40       # trap 'process_shm --id $NVMF_APP_SHM_ID; nvmftestfini; exit 1' SIGINT SIGTERM EXIT
 14:58:40       # ip netns exec ens786f0_ns /home/qwan/CVL_Nightly/spdk/build/bin/nvmf_tgt -i 0 -e 0xFFFF -m 0xF
 14:58:40       # waitforlisten 59291
 14:58:40       # '[' -z 59291 ']'
 14:58:40       # local rpc_addr=/var/tmp/spdk.sock
 14:58:40       # echo 'Waiting for process to start up and listen on UNIX domain socket /var/tmp/spdk.sock...'
Waiting for process to start up and listen on UNIX domain socket /var/tmp/spdk.sock...
 14:58:40       # xtrace_disable
 14:58:40       # '[' no '!=' yes ']'
 14:58:40       # PREV_BASH_OPTS=ehxBET
 14:58:40       # [[ ehxBET == *\x* ]]
 14:58:40       # XTRACE_DISABLED=yes
 14:58:40       # set +x
[2021-04-23 14:58:40.439425] Starting SPDK v21.04-pre git sha1 ec94874c1 / DPDK 20.08.0 initialization...
[2021-04-23 14:58:40.439525] [ DPDK EAL parameters: [2021-04-23 14:58:40.439538] nvmf [2021-04-23 14:58:40.439546] -c 0xF [2021-04-23 14:58:40.439552] --log-level=lib.eal:6 [2021-04-23 14:58:40.439559] --log-level=lib.cryptodev:5 [2021-04-23 14:58:40.439566] --log-level=user1:6 [2021-04-23 14:58:40.439573] --base-virtaddr=0x200000000000 [2021-04-23 14:58:40.439579] --match-allocations [2021-04-23 14:58:40.439586] --file-prefix=spdk0 [2021-04-23 14:58:40.439594] --proc-type=auto [2021-04-23 14:58:40.439601] ]
EAL: No available hugepages reported in hugepages-2048kB
EAL: No free hugepages reported in hugepages-1048576kB
EAL: No legacy callbacks, legacy socket not created
[2021-04-23 14:58:40.771920] app.c: 535:spdk_app_start: *NOTICE*: Total cores available: 4
[2021-04-23 14:58:40.931559] app.c: 383:app_setup_trace: *NOTICE*: Tracepoint Group Mask 0xFFFF specified.
[2021-04-23 14:58:40.931671] app.c: 387:app_setup_trace: *NOTICE*: Use 'spdk_trace -s nvmf -i 0' to capture a snapshot of events at runtime.
[2021-04-23 14:58:40.931687] app.c: 389:app_setup_trace: *NOTICE*: Or copy /dev/shm/nvmf_trace.0 for offline analysis/debug.
[2021-04-23 14:58:40.931802] reactor.c: 911:reactor_run: *NOTICE*: Reactor started on core 1
[2021-04-23 14:58:40.931869] reactor.c: 911:reactor_run: *NOTICE*: Reactor started on core 3
[2021-04-23 14:58:40.931871] reactor.c: 911:reactor_run: *NOTICE*: Reactor started on core 0
[2021-04-23 14:58:40.931821] reactor.c: 911:reactor_run: *NOTICE*: Reactor started on core 2
[2021-04-23 14:58:40.932042] accel_engine.c: 854:spdk_accel_engine_initialize: *NOTICE*: Accel engine initialized to use software engine.
 14:58:41       # xtrace_enable
 14:58:41       # (( i == 0 ))
 14:58:41       # return 0
 14:58:41       # /home/qwan/CVL_Nightly/spdk/scripts/rpc.py nvmf_create_transport -t tcp -o -u 8192
[2021-04-23 14:58:41.289872] tcp.c: 554:nvmf_tcp_create: *NOTICE*: *** TCP Transport Init ***
 14:58:41       # timing_exit start_nvmf_tgt
 14:58:41       # xtrace_disable
 14:58:41       # '[' no '!=' yes ']'
 14:58:41       # PREV_BASH_OPTS=ehxBET
 14:58:41       # [[ ehxBET == *\x* ]]
 14:58:41       # XTRACE_DISABLED=yes
 14:58:41       # set +x
 14:58:41       # xtrace_enable
 14:58:41       # /home/qwan/CVL_Nightly/spdk/scripts/rpc.py bdev_malloc_create 64 512 -b Malloc1
Malloc1
 14:58:41       # /home/qwan/CVL_Nightly/spdk/scripts/rpc.py nvmf_create_subsystem nqn.2016-06.io.spdk:cnode1 -a -s SPDK00000000000001
 14:58:41       # /home/qwan/CVL_Nightly/spdk/scripts/rpc.py nvmf_subsystem_add_ns nqn.2016-06.io.spdk:cnode1 Malloc1
 14:58:41       # /home/qwan/CVL_Nightly/spdk/scripts/rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a 10.0.0.2 -s 4420
[2021-04-23 14:58:41.800600] tcp.c: 749:nvmf_tcp_listen: *NOTICE*: *** NVMe/TCP Target Listening on 10.0.0.2 port 4420 ***
 14:58:41       # PLUGIN_DIR=/home/qwan/CVL_Nightly/spdk/examples/nvme/fio_plugin
 14:58:41       # fio_nvme /home/qwan/CVL_Nightly/spdk/examples/nvme/fio_plugin/example_config.fio '--filename=trtype=tcp adrfam=IPv4 traddr=10.0.0.2 trsvcid=4420 ns=1' --bs=4096
 14:58:41       # local fio_dir=/usr/src/fio
 14:58:41       # local nvme_plugin=/home/qwan/CVL_Nightly/spdk/build/fio/spdk_nvme
  14:58:41      # grep libasan
  14:58:41      # ldd /home/qwan/CVL_Nightly/spdk/build/fio/spdk_nvme
  14:58:41      # awk '{print $3}'
 14:58:41       # asan_lib=
 14:58:41       # LD_PRELOAD=' /home/qwan/CVL_Nightly/spdk/build/fio/spdk_nvme'
 14:58:41       # /usr/src/fio/fio /home/qwan/CVL_Nightly/spdk/examples/nvme/fio_plugin/example_config.fio '--filename=trtype=tcp adrfam=IPv4 traddr=10.0.0.2 trsvcid=4420 ns=1' --bs=4096
test: (g=0): rw=randrw, bs=(R) 4096B-4096B, (W) 4096B-4096B, (T) 4096B-4096B, ioengine=spdk, iodepth=128
fio-3.7
Starting 1 thread
[2021-04-23 14:58:42.224229] Starting SPDK v21.04-pre git sha1 ec94874c1 / DPDK 20.08.0 initialization...
[2021-04-23 14:58:42.224317] [ DPDK EAL parameters: [2021-04-23 14:58:42.224376] fio [2021-04-23 14:58:42.224387] --no-shconf [2021-04-23 14:58:42.224396] -c 0x1 [2021-04-23 14:58:42.224404] -m 0 [2021-04-23 14:58:42.224411] --log-level=lib.eal:6 [2021-04-23 14:58:42.224418] --log-level=lib.cryptodev:5 [2021-04-23 14:58:42.224426] --log-level=user1:6 [2021-04-23 14:58:42.224433] --base-virtaddr=0x200000000000 [2021-04-23 14:58:42.224440] --match-allocations [2021-04-23 14:58:42.224448] --file-prefix=spdk_pid59336 [2021-04-23 14:58:42.224456] ]
EAL: No available hugepages reported in hugepages-2048kB
EAL: No free hugepages reported in hugepages-1048576kB
EAL: No legacy callbacks, legacy socket not created
[2021-04-23 14:59:32.953875] ctrlr.c: 186:nvmf_ctrlr_keep_alive_poll: *NOTICE*: Disconnecting host nqn.2014-08.org.nvmexpress:uuid:e5b9f98e-409c-4633-a561-598223142fe8
 from subsystem nqn.2016-06.io.spdk:cnode1 due to keep alive timeout.
Jobs: 1 (f=1): [m(1)][100.0%][r=0KiB/s,w=0KiB/s][r=0,w=0 IOPS][eta 00m:00s]

```

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @optimistyzy. log: https://ci.spdk.io/public_build/autotest-per-patch_43272.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @tomzawadzki. log: https://ci.spdk.io/public_build/autotest-per-patch_43359.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @mikeBashStuff. log: https://ci.spdk.io/public_build/autotest-per-patch_43587.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @wanghailiangx. log: https://ci.spdk.io/public_build/autotest-per-patch_43667.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @shuhei-matsumoto. log: https://ci.spdk.io/public_build/autotest-per-patch_43715.html

**Comment by wanghailiangx** [comment]
@shuhei-matsumoto Please see https://github.com/spdk/spdk/issues/1918, thanks, your above report is nothing about issues #1906.

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @mikeBashStuff. log: https://ci.spdk.io/public_build/autotest-per-patch_46988.html

**Comment by tomzawadzki** [comment]
@wanghailiangx have you have a chance to take a look at the issue and patch working around it ?

**Comment by wanghailiangx** [comment]
@tomzawadzki I haven’t seen this issue reappear in the last month, could you close it?
The NetworkManager.service was stopped in CI, this method fixed such a series of issues #1906, #1900.
